core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/OnCompletionForwarder.java

```diff
@@ -0,0 +1,55 @@
+package com.novoda.noplayer.exoplayer.forwarder;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.PlaybackParameters;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.novoda.noplayer.listeners.CompletionListeners;
+
+class OnCompletionForwarder implements ExoPlayer.EventListener {
+
+    private final CompletionListeners completionListeners;
+
+    OnCompletionForwarder(CompletionListeners completionListeners) {
+        this.completionListeners = completionListeners;
+    }
+
+    @Override
+    public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+        if (playbackState == ExoPlayer.STATE_ENDED) {
+            completionListeners.onCompletion();
+        }
+    }
+
+    @Override
+    public void onTimelineChanged(Timeline timeline, Object manifest) {
+        //TODO should we send ?
```

Comment:
Again, we need to figure out the mapping from `ExoPlayer` to `no-player` for these additional events.